### DISCUSSION TITLE
Why is `.withReuse(true)` not the default?

### DISCUSSION THREAD

**Discussion by wimdeblauwe** [testcontainers/testcontainers-java#5023] [Q&A] [answered]
I wonder why there are 2 things you need to do to get container re-use: set the property in your home directory + add `.withReuse(true)` in the Java code itself. It seems like this could be the default as it would only kick in if you have that property in your home directory anyway. Or are there potential pitfalls I need to be aware of that made you decide _not_ to make that the default?

**Comment by kiview** [comment, marked as answer]
Hey @wimdeblauwe, happy to hear that you seem to like `withReuse` so much, you could imagine it being the default 🙂 

Reusable containers is still an alpha feature with only experimental support, so we currently want to make the opt-in very explicit. In addition, `withReuse` indicates if a container _can_ be reused in the context of its creation (or its test), while the flag in `.testcontainers.properties` indicates that the runtime environment would benefit from reuse (e.g. we generally don't want CI to use reuse). 

Still thanks for your comment, we will consider the different UX aspects once we move forward in making this feature GA 👍

**Reply by wimdeblauwe** [reply]
Thanks for clarifying this. To be clear: I have not used it yet, because the fact it is not the default made me hesitant to start using it. But as you point out, as long as the CI does not have that properties file, there is nothing that can go wrong as it would not kick in anyway there.

**Comment by bsideup** [comment]
Not every container is reusable. For example, if you set an environment variable to something random, it will always start a new container.

`withReuse` is your "opt-in" way of saying that "yes, X *can* be reused", and the property in `testcontainers.properties` says that "yes, this environment *should* reuse containers".
